misc/dbt-materialize/tests/adapter/test_utils.py

```diff
@@ -0,0 +1,136 @@
+from dbt.tests.adapter.utils.test_datediff import BaseDateDiff
+from dbt.tests.adapter.utils.test_date_trunc import BaseDateTrunc
+from dbt.tests.adapter.utils.test_escape_single_quotes import BaseEscapeSingleQuotesQuote
+from dbt.tests.adapter.utils.test_escape_single_quotes import BaseEscapeSingleQuotesBackslash
+from dbt.tests.adapter.utils.test_except import BaseExcept
+from dbt.tests.adapter.utils.test_hash import BaseHash
+from dbt.tests.adapter.utils.test_intersect import BaseIntersect
+from dbt.tests.adapter.utils.test_last_day import BaseLastDay
+from dbt.tests.adapter.utils.test_length import BaseLength
+from dbt.tests.adapter.utils.test_listagg import BaseListagg
+from dbt.tests.adapter.utils.test_position import BasePosition
+from dbt.tests.adapter.utils.test_replace import BaseReplace
+from dbt.tests.adapter.utils.test_right import BaseRight
+from dbt.tests.adapter.utils.test_safe_cast import BaseSafeCast
+from dbt.tests.adapter.utils.test_split_part import BaseSplitPart
+from dbt.tests.adapter.utils.test_string_literal import BaseStringLiteral
+from dbt.tests.adapter.utils.fixture_cast_bool_to_text import models__test_cast_bool_to_text_yml
+
+
+models__test_cast_bool_to_text_sql = """
+with data_bool as (
+    select 0=1 as input, 'false' as expected union all
+    select 1=1 as input, 'true' as expected
+),
+
+data_null as (
+    select null as input, null as expected
+)
+
+select
+    {{ cast_bool_to_text("input") }} as actual,
+    expected
+from data_bool
+UNION ALL
+select
+    {{ cast_bool_to_text("input") }} as actual,
+    expected
+from data_null
+"""
+
+
+class TestAnyValue(BaseAnyValue):
+    pass
+
+@pytest.mark.skip(reason="Materialize does not yet support bool_or() see https://github.com/MaterializeInc/materialize/issues/3154")
+class TestBoolOr(BaseBoolOr):
+    pass
+
+
+# The cast_bool_to_text macro works as expected. We alter the test case to accommodate.
+# Set conversion type operations issue: https://github.com/MaterializeInc/materialize/issues/3331
+class TestCastBoolToText(BaseCastBoolToText):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "test_cast_bool_to_text.yml": models__test_cast_bool_to_text_yml,
+            "test_cast_bool_to_text.sql": self.interpolate_macro_namespace(
+                models__test_cast_bool_to_text_sql, "cast_bool_to_text"
+            ),
+        }
+    pass
+
+
+class TestConcat(BaseConcat):
+    pass
+
+
+class TestDateAdd(BaseDateAdd):
+    pass
+
+
+class TestDateDiff(BaseDateDiff):
+    pass
+
+
+class TestDateTrunc(BaseDateTrunc):
+    pass
+
+
+class TestEscapeSingleQuotes(BaseEscapeSingleQuotesQuote):
+    pass
+
+
+class TestExcept(BaseExcept):
+    pass
+
+
+class TestHash(BaseHash):
+    pass
+
+
+class TestIntersect(BaseIntersect):
+    pass
+
+
+class TestLastDay(BaseLastDay):
+    pass
+
+
+class TestLength(BaseLength):
+    pass
+
+## TODO: Materialize list_agg() does not take a delimiter.
```

Comment:
We have two options here: 
1. Update the dbt tests to reflect what actually happens when list_agg() is invoked in Materialize.
2. Create a `materialize__listagg()` macro which adheres to the standard used across databases, similar to the [postgres](https://github.com/dbt-labs/dbt-core/blob/main/plugins/postgres/dbt/include/postgres/macros/utils/listagg.sql) version.

Reply:
I went with a third option. `list_agg()` in materialize does not take a delimiter, and it does not return a string. This functionality differs enough from the base dbt implementation of listagg that i've skipped including this macro, for now, in favor of users just implementing list_agg() themselves. We did not have list_agg supported in materialize-dbt-utils, so this will not break anything.